DiceKit/SubtractionExpression.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  SubtractionExpression.swift
+//  DiceKit
+//
+//  Created by Jonathan Hoffman on 8/15/15.
+//  Copyright © 2015 Brentley Jones. All rights reserved.
+//
+
+import Foundation
+
+public struct SubtractionExpression<LeftExpression: protocol<ExpressionType, Equatable>, RightExpression: protocol<ExpressionType, Equatable>>: Equatable {
+    
+    public let minuend: LeftExpression
+    public let subtrahend: RightExpression
+    
+    public init(_ minuend: LeftExpression, _ subtrahend: RightExpression) {
+        self.minuend = minuend
+        self.subtrahend = subtrahend
+    }
+    
+}
+
+// MARK: - ExpressionType
+
+extension SubtractionExpression: ExpressionType {
+    
+    public typealias Result = SubtractionExpressionResult<LeftExpression.Result, RightExpression.Result>
+    
+    public func evaluate() -> Result {
+        let leftResult = minuend.evaluate()
+        let rightResult = subtrahend.evaluate()
+        
+        return Result(leftResult, rightResult)
+    }
+    
+    public var probabilityMass: ExpressionProbabilityMass {
+        return minuend.probabilityMass && NegationExpression(subtrahend).probabilityMass
```

Comment:
`minuend.probabilityMass && subtrahend.probabilityMass.negateOutcomes` would be preferable, since it doesn't add a dependency on another expression.